### PR TITLE
Add logs during legacy migration

### PR DESCRIPTION
**Pull request by h3nrique**
Added informational logs and Included steps do skip creation of 'account' client during legacy migration database.

### CODE REVIEW

File: model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java [hunk 1]
```diff
@@ -220,8 +220,9 @@ public RoleModel addRealmRole(RealmModel realm, String name) {
     }
     @Override
     public RoleModel addRealmRole(RealmModel realm, String id, String name) {
+        logger.tracef("addRealmRole(%s, %s, %s)", realm.getId(), id, name);
```
Comment (hmlnarik) [inline model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java:223]:
```suggestion
        logger.tracef("addRealmRole(%s, %s, %s)%s", realm, id, name, getShortStackTrace());
```

File: model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java [hunk 2]
```diff
@@ -251,8 +252,9 @@ public RoleModel addClientRole(ClientModel client, String name) {
 
     @Override
     public RoleModel addClientRole(ClientModel client, String id, String name) {
+        logger.tracef("addClientRole(%s, %s, %s)", client.getId(), id, name);
```
Comment (hmlnarik) [inline model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java:255]:
```suggestion
        logger.tracef("addClientRole(%s, %s, %s)%s", client, id, name, getShortStackTrace());
```

**Comment by hmlnarik** [comment]
Thank you for the PR.
For it to be reviewed, please ensure that you have read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md before submitting a PR, and provide proper bug report including existing and expected behaviour to set the context of the change.

File: model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java [hunk 3]
```diff
@@ -92,8 +97,13 @@ protected void addAccountConsoleClient(RealmModel realm) {
...
 
-            RoleModel role = realm.getClientByClientId(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID).getRole(AccountRoles.MANAGE_ACCOUNT);
-            if (role != null) client.addScopeMapping(role);
+            ClientModel accountClient = realm.getClientByClientId(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID);
+            if (accountClient != null) {
+                RoleModel role = accountClient.getRole(AccountRoles.MANAGE_ACCOUNT);
+                if (role != null) {
+                    client.addScopeMapping(role);
+                }
+            }
```
Comment (hmlnarik) [inline model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java:106]: This change may or may not be valid. File a bug report with justification before this PR.

**Review by hmlnarik — CHANGES_REQUESTED** [review]
See above

File: model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java (hunk 3 above)
Comment (stianst) [reply model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java:106]: This really should be split into two issues/PRs

Comment (h3nrique) [reply model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java:106]: Hi Hynek,

In fact, there is no bug reported for this change.

However, will occur a NPE if the keycloak administrator had removed the client 'account'. I can open a bug for this if you think better.

I believe, based on my observation, that the existence of this client should not interrupt the migration.

Comment (h3nrique) [reply model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java:106]: > This really should be split into two issues/PRs

Ok

Comment (h3nrique) [reply model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java:106]: Only for tracing.

#15020

**Comment by hmlnarik** [comment]
@h3nrique Please do follow contributor's guide and kindly refrain from resolving the comments yourself, they will be resolved by the reviewer after they have been addressed and reviewer agrees to their resolution.

The review comments have not been addressed so far.

**Comment by hmlnarik** [comment]
@h3nrique Could you please let me know if you plan to continue working on this issue

**Comment by h3nrique** [comment]
> @h3nrique Could you please let me know if you plan to continue working on this issue

No, I finished.

**Comment by hmlnarik** [comment]
> > @h3nrique Could you please let me know if you plan to continue working on this issue
> 
> No, I finished.

Closing since the review comments were not addressed and the author discontinued the PR.

**Pull request closed without merging**